Feature/Checkbox ListItem DAO-615

RakeshUP opened the pull request.

Fabricevladimir commented on packages/ui-components/src/components/checkbox/checkboxListItem.tsx:
```diff
@@ -0,0 +1,93 @@
+  multiSelect,
+  isActive = false,
+  disabled = false,
+  isMulti = false,
```
Can you explain a bit what this prop is for? It's a bit ambiguous especially with the other prop `multiselect`

sepehr2github replied:
There is no onClick or onChange implemented so when I click on the list item nothing happened!. And I think we should have a better way of handling the props. So based on the design we have two types (multi-selection: yes or no) and we have three states for multi-selection (default/active/multi) and two for single selection (default/active) all status would change by user clicks

Fabricevladimir commented on packages/ui-components/src/components/checkbox/checkboxListItem.tsx:
```diff
@@ -0,0 +1,93 @@
+  },
+};
+
+export type CheckboxListItemProps = {
```
How do we toggle the state of the checkbox?

Fabricevladimir reviewed (commented):
Good work, and thanks for fixing the circular dependency issue. Left a couple of questions.

sepehr2github commented on packages/ui-components/src/components/checkbox/checkboxListItem.tsx:
```diff
@@ -0,0 +1,93 @@
+      disabled={disabled}
+      isMulti={isMulti}
+    >
+      <HStack isActive={isActive} disabled={disabled} isMulti={isMulti}>
```
and also can you add cursor: pointer for hover?

sepehr2github reviewed (commented):
Thank you! Rakash. I left a few comments for you. 👍

RakeshUP commented:
@Fabricevladimir @sepehr2github I've added onclick prop and a controlled way to toggle the state inside the component itself. I've changed the other ambiguous props too

Fabricevladimir reviewed (approved):
Looks great! Thanks
